Dockerfile

```diff
@@ -242,6 +243,12 @@ RUN if [[ -n $TF_VERSION ]]; then \
         fi; \
     fi
 
+# install ONNX Runtime
+ARG ONNX_RUNTIME_VERSION
+RUN if [[ -n $ONNX_RUNTIME_VERSION ]]; then \
+        pip3 install onnxruntime-gpu==${ONNX_RUNTIME_VERSION}; \
```

Comment:
Test ONNX RT, see https://github.com/ika-rwth-aachen/docker-ros-ml-images/issues/11#issuecomment-2497455521.

Reply:
[This sample script](https://github.com/microsoft/onnxruntime-inference-examples/blob/main/python/api/onnxruntime-python-api.py) could be used for testing.

It currently fails with an error that `libcudnn.so.9` is not found. According to [the docs](https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#cuda-12x), the newest ONNX-RT version requires cuDNN 9.x. We are still on cuDNN 8.x though.

Downgrading to `onnxruntime-gpu~=1.8.0` (for cuDNN 8.x), however, throws a similar error for `libcublas`, where we have v12 installed instead of v11.

```
2024-11-25 20:33:08.247359117 [E:onnxruntime:Default, provider_bridge_ort.cc:1745 TryGetProviderInfo_CUDA] /onnxruntime_src/onnxruntime/core/session/provider_bridge_ort.cc:1426 onnxruntime::Provider& onnxruntime::ProviderLibrary::Get() [ONNXRuntimeError] : 1 : FAIL : Failed to load library libonnxruntime_providers_cuda.so with error: libcublasLt.so.11: cannot open shared object file: No such file or directory
```

Reply:
The sample script mentioned above is successful in a CUDA 12.6 `ros2-torch:humble` image.

```
pip3 install onnx==1.17.0 onnxruntime-gpu==1.20.1
```

We are soon going to update to CUDA 12.6, afterwards this ONNX PR should be pretty straightforward.

Reply:
Can be tackled after #14 has been merged.